src/Compilers/Core/Portable/Emit/EditAndContinue/DeltaMetadataWriter.cs

```diff
@@ -784,6 +786,60 @@ private void PopulateEncLogTableParameters()
             }
         }
 
+        private void PopulateEncLogTableCustomAttributes()
+        {
+            var attributeMap = CreateExistingAttributeMap(_previousGeneration.MetadataReader);
+
+            var nextCustomAttributeRow = _previousGeneration.MetadataReader.GetTableRowCount(TableIndex.CustomAttribute) + 1;
+
+            foreach (var customAttributeTarget in _customAttributeTargets)
+            {
+                int row;
+                if (attributeMap.TryGetValue(customAttributeTarget, out var queue) &&
+                    queue.Count > 0)
+                {
+                    // If there is already an attribute for this parent, then pull the next available row number
+                    // off the queue and update it
+                    row = queue.Dequeue();
```

Comment:
>row = queue.Dequeue() [](http://example.com/codeflow?start=20&length=21)

How do we know we're updating the correct attribute row? What if this particular attribute was deleted in the EnC update?

Reply:
> How do we know we're updating the correct attribute row?

This is deliberately not trying to update the "correct" attribute, it just emits all of them all of the time, which was the existing behavior anyway, it just always did them as new rows. Right now to do anything smarter would require the metadata writer to duplicate a bunch of logic that the EnC Analyzer is doing to detect updates, adds, moves etc. because without symbols for attributes, the analyzer can't send edits for specific attributes to inform what needs updating.

> What if this particular attribute was deleted in the EnC update?

Deletes I still need to do, but my plan is essentially to zero-out any CustomAttribute rows for this target that are no longer needed (ie, anything left in any queue)

Reply:
> Deletes I still need to do, but my plan is essentially to zero-out any CustomAttribute rows for this target that are no longer needed (ie, anything left in any queue)

Update: This plan is likely to be abandoned!